src/sciline/data_graph.py

```diff
@@ -250,6 +270,40 @@ def visualize_data_graph(self, **kwargs: Any) -> graphviz.Digraph:
         return dot
 
 
+class GroupbyGraph:
+    """
+    A graph that has been grouped by a specific index.
+    This is a specialized graph that is used to represent the result of a groupby
+    operation. It allows for operations on the grouped data,
+    such as aggregation or summarization.
+    """
+
+    def __init__(
+        self, graph: CbGroupbyGraph, graph_maker: Callable[..., DataGraph]
+    ) -> None:
+        self._cbgraph = graph
+        # We forward the constructor so this can be used by both DataGraph and Pipeline
+        self._graph_maker = graph_maker
+
+    def reduce(self, *, func: Callable[..., Any], **kwargs: Any) -> DataGraph:
```

Comment:
Since this is user-facing we should repeat all the args (for auto-complete) and docstrings, instead of just using kwargs.